qiime/sdk/tests/test_method.py

```diff
@@ -374,6 +376,93 @@ def test_async(self):
             self.assertEqual(result.view(list),
                              [10, 20, 0, 42, 43, 99, -22, 55, 1])
 
+    def test_markdown_input_section_validation(self):
+        temp_dir = tempfile.TemporaryDirectory(prefix="qiime2-test-temp-")
+        root_dir = temp_dir.name
+        fp = os.path.join(root_dir, "bad_markdown.md")
+
+        with open(fp, mode="w") as fh:
+            fh.write("""---
+name: Bad markdown
+description: "incorrectly formatted inputs"
+inputs:
+    - ints1:
+        - IntSequence1 | IntSequence2
+        - list
+        - Oops
+parameters:
+    - int1:
+        - Int
+        - int
+outputs:
+    - concatenated_ints:
+        - IntSequence1
+        - list
+---""")
+        with self.assertRaises(TypeError) as te:
```

Comment:
It'd be preferable to use

```python
self.assertRaisesRegex(TypeError, 'input section'):
    ...
```

so that the test case isn't too dependent on the exact error message that's raised, in case that changes in the future. I recommend using a similar regex for the other error cases (e.g. `'parameters section'`, `'outputs section'`).